### PR TITLE
Cover Admin::AccountsController more

### PR DESCRIPTION
**Pull request by akihikodaki**

### CODE REVIEW

**Review by mjankowski — DISMISSED** [review]
A lot of these are just duplicating specs for the filter ... and would not fail if the filter was removed.

**Comment by akihikodaki** [comment]
@mjankowski 
> and would not fail if the filter was removed.

It would.

```console
$ bundle exec rspec spec/controllers/admin/accounts_controller_spec.rb
[DEPRECATION] `have_enqueued_job` is deprecated.  Please use `have_enqueued_sidekiq_job` instead.

Randomized with seed 3870
                                                                                                                                                      
  1) Admin::AccountsController GET #index filters with local parameter
     Failure/Error: expect(accounts.size).to eq 1
     
       expected: 1
            got: 2
     
       (compared using ==)
     # ./spec/controllers/admin/accounts_controller_spec.rb:20:in `block (3 levels) in <top (required)>'

 13/13 |=========================================================== 100 ============================================================>| Time: 00:00:03 

Finished in 3.35 seconds (files took 2.37 seconds to load)
13 examples, 1 failure

Failed examples:

rspec ./spec/controllers/admin/accounts_controller_spec.rb:13 # Admin::AccountsController GET #index filters with local parameter

Randomized with seed 3870

Coverage report generated for RSpec to /home/akihikodaki/go/src/github.com/tootsuite/mastodon/coverage. 500 / 8387 LOC (5.96%) covered.
$ git diff
diff --git a/app/controllers/admin/accounts_controller.rb b/app/controllers/admin/accounts_controller.rb
index 4c41c421..680ee73a 100644
--- a/app/controllers/admin/accounts_controller.rb
+++ b/app/controllers/admin/accounts_controller.rb
@@ -18,7 +18,7 @@ module Admin
 
     def filter_params
       params.permit(
-        :local,
+        #:local,
         :remote,
         :by_domain,
         :silenced,
```

**Comment by akihikodaki** [comment]
80822dbb2af321bc39753a027a91a8eb20642fa8 mocks `AccountFilter.new` rather than calling it so many times.

File: spec/controllers/admin/accounts_controller_spec.rb [hunk 1]
```diff
@@ -3,11 +3,57 @@
...
+        suspended: '1',
+        username: 'username',
+        display_name: 'display name',
+        email: 'local-part@domain',
+        ip: '0.0.0.42'
+      }
+    end
+
+    it 'paginates accounts' do
+      40.times { Fabricate(:account) }
```
Comment (mjankowski) [inline spec/controllers/admin/accounts_controller_spec.rb]: Same concern here as the other one...

**Comment by akihikodaki** [comment]
27caa560b6787039f4296ebb0c37e24fd61b6bd9 limits entries per page.

**Comment by akihikodaki** [comment]
85b6cf73984691c1a3a756ae0c12920ef21a86e3 applies https://github.com/tootsuite/mastodon/pull/3329#discussion_r118766156.

**Comment by akihikodaki** [comment]
Any update?

**Review by Gargron — APPROVED** [review]

**Pull request merged**